component-tests/rust/src/main.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+mod common;
```

Comment:
```suggestion
// SPDX-License-Identifier: Apache-2.0 OR MIT

#![warn(clippy::alloc_instead_of_core)]
#![warn(clippy::std_instead_of_alloc)]
#![warn(clippy::std_instead_of_core)]

mod common;
```

We also need to run these tests on the target platform, which might not support `std`. While these tests do not need to be `no_std` right away, we should try to use the `core` and `alloc` crates if the type is re-exported from there.

You just need to replace `std::fmt` with `core::fmt`.